### ISSUE TITLE
Update outdated node modules to latest versions

### ISSUE THREAD

**Issue by lewisdiamond** [arangodb/arangodb#2138]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [x] 3.0
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [x] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [x] other: Archlinux -> Docker -> (Debian?)
- [ ] Windows, version: 
- [ ] MacOS, version:

Mocha, for example, is quite broken. beforeEach/afterEach runs before/after every single test defined, not just those on the same block level. I.e. it will run on tests for completely unrelated modules and break them.

**Comment by pluma** [comment]
The mocha problems have been addressed in recent commits to devel, 3.1 and 3.0 and should be fixed.

The outdated packages still need to be addressed though.

**Comment by pluma** [comment]
Note to myself: Output from `npm --outdated`:
```
Package                     Current  Wanted  Latest  Location
eslint                       2.13.1  2.13.1   3.9.1
eslint-config-semistandard    6.0.2   6.0.2   7.0.0
eslint-config-standard        5.3.1   5.3.1   6.2.1
eslint-plugin-promise         1.3.2   1.3.2   3.3.0
eslint-plugin-standard        1.3.2   1.3.2   2.0.1
highlight.js                  9.4.0   9.4.0   9.8.0
joi                           8.4.2   8.4.2   9.2.0
joi-to-json-schema            2.2.0   2.2.0   2.3.0
lodash                       4.13.1  4.13.1  4.16.6
marked                        0.3.5   0.3.5   0.3.6
mime-types                   2.1.11  2.1.11  2.1.12
minimatch                     3.0.0   3.0.0   3.0.3
mocha                         2.5.3   2.5.3   3.1.2
qs                            6.2.0   6.2.0   6.3.0
semver                        5.1.0   5.1.0   5.3.0
sinon                        1.17.4  1.17.4  1.17.6
```

I'm not sure why all the eslint stuff is in there. I thought we are now using semistandard, which also comes as a standalone module? We should check joi for breaking changes as well as what changed between mocha 2.5 and 3.1 (we can't simply update mocha via npm).
